hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/MemStoreCompactor.java

```diff
@@ -98,17 +102,19 @@ public MemStoreCompactor(CompactingMemStore compactingMemStore) {
    * is already an ongoing compaction or no segments to compact.
    */
   public boolean start() throws IOException {
-    if (!compactingMemStore.hasImmutableSegments()) return false;  // no compaction on empty
+    if (!compactingMemStore.hasImmutableSegments()) // no compaction on empty pipeline
+      return false;
 
-    int t = compactingMemStore.getConfiguration().getInt(COMPACTING_MEMSTORE_TYPE_KEY,
+    String memStoreType = compactingMemStore.getConfiguration().get(COMPACTING_MEMSTORE_TYPE_KEY,
         COMPACTING_MEMSTORE_TYPE_DEFAULT);
 
-    switch (t) {
-      case 1: type = Type.COMPACT_TO_SKIPLIST_MAP;
+    switch (memStoreType) {
+      case "index-compaction": action = Action.MERGE;
```

Comment:
If we already assign the right names to the configuration parameters, why not call the enum's accordingly?